src/app/shared/models/token.ts

```diff
@@ -0,0 +1,4 @@
+export type Token = {
```

Comment:
Find a better name for this type. You say it's `Token`, but it contains TWO different tokens